src/Container.php

```diff
@@ -397,7 +397,7 @@ protected function getRaw($key)
 
 		$aliasKey = $this->resolveAlias($key);
 
-		if ($aliasKey != $key && isset($this->dataStore[$aliasKey]))
+		if ($aliasKey !== $key && isset($this->dataStore[$aliasKey]))
```

Comment:
As long as it's not forcing case sensitive matches then we're good.  I just assumed strict comparisons might take casing into consideration, but apparently it doesn't.